source/Camera/Basic/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,106 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string &filename) {
+	model.WriteObjectModel3d(HalconCpp::HString{ "ply" },
+		HalconCpp::HString{ filename.c_str() },
+		HalconCpp::HString{ "invert_normals" },
+		HalconCpp::HString{ "false" });
+}
+
+
+void setColorsInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, const HalconCpp::HImage &RGB) {
+
+	const HalconCpp::HRegion domain = RGB.AccessChannel(1).GetDomain();
+	HalconCpp::HTuple rows, cols;
+	domain.GetRegionPoints(&rows, &cols);
+
+	const HalconCpp::HTuple Rtup = RGB.AccessChannel(1).GetGrayval(rows, cols);
+	const HalconCpp::HTuple Gtup = RGB.AccessChannel(2).GetGrayval(rows, cols);
+	const HalconCpp::HTuple Btup = RGB.AccessChannel(3).GetGrayval(rows, cols);
+
+	std::cout << "Adding RGB to ObjectModel3D" << std::endl;
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+
+}
+
+
+HalconCpp::HString getFirstAvailableZividDevice(const HalconCpp::HTuple &devices) {
+	auto zividDevices = devices.TupleRegexpSelect("Zivid");
+
+	if (zividDevices.Length() == 0) {
+		std::cout << "No Zivid devices found. Please check your setup" << std::endl;
+	}
```

Comment:
If not cameras are conencted thow an error?
`if (zividDevices.Length() == 0) throw std::runtime_error("No Zivid devices found. Please check your setup.");`

Reply:
done